deployer/helm_upgrade_decision.py

```diff
@@ -0,0 +1,368 @@
+"""
+Functions related to deciding which clusters and/or hubs need their *hub helm chart or
+support helm chart upgrading depending on an input list of filenames that have been
+added or modified in a GitHub Pull Request.
+"""
+import fnmatch
+import os
+from pathlib import Path
+from ruamel.yaml import YAML
+from rich.console import Console
+from rich.table import Table
+
+yaml = YAML(typ="safe", pure=True)
+
+# Determine if we are running a test or not. We set this env var to "test" in the
+# pyproject.toml file so it is set when the package is tested using pytest.
+test_env = os.getenv("RUN_ENV", False)
+
+
+def discover_modified_common_files(modified_paths: list):
+    """There are certain common files which, if modified, we should upgrade all hubs
+    and/or all clusters appropriately. These common files include the helm charts we
+    deploy, as well as the GitHub Actions and deployer package we use to deploy them.
+
+    Args:
+        modified_paths (list[str]): The list of files that have been added or modified
+            in a given GitHub Pull Request.
+
+    Returns:
+        upgrade_support_on_all_clusters (bool): Whether or not all clusters should be
+            upgraded since the support chart has changed
```

Comment:
```suggestion
        upgrade_support_on_all_clusters (bool): Whether or not all clusters should have
            their support chart upgraded since has changes
```